packages/data-service/src/data-service.ts

```diff
@@ -937,7 +958,19 @@ export class DataServiceImpl extends EventEmitter implements DataService {
 
     const listCollections = async () => {
       try {
-        return await db.listCollections(filter, { nameOnly }).toArray();
+        const cursor = db.listCollections(filter, { nameOnly });
+        // Iterate instead of using .toArray() so we can emit
+        // collection info update events as they come in.
+        const results = [];
+        for await (const result of cursor) {
+          this.emit(
```

Comment:
It can, but I’m also not quite sure that I understand the concern. The goal here is to keep the csfle collection tracking implementation as self-contained as possible, without the DataService class having to know about its internals.

Reply:
Ah yeah makes sense actually the listener is in a different class, we want to refresh a "cache" in the csfle collection tracking and the collection tracking has a reference to the dataService, not the other way around, I was hallucinating sorry.

What about the event name? Can we use something like `collection-info-fetched`, or similar to reflect that is an event on "read"?

Reply:
> Can we use something like `collection-info-fetched`, or similar to reflect that is an event on "read"?

Sure, that sounds good! Done :+1: (along with changes to the CI, now running this inside a separate job because the setup work turned out to be a bit more complicated than I’d have liked)